### ISSUE TITLE
Reverse Proxy HTTPS Domain Fail To Play

### ISSUE THREAD

**Issue by Nazar78** [jellyfin/jellyfin-tizen#46]
Hi,

I've configured JF 10.6.2 to run behind Nginx reverse proxy with tmpfs ram caching only for HTTP 200 GET method. I've no issues so far with browsers and Android clients however with Tizen 5 both TV and emulator, I randomly have issues trying to play any video. Upon clicking the play or resume button, the backdrop loads followed by the circular loading spinning endlessly. There's no particular errors in the JF nor Nginx logs, also there's no further request seen in the JF log for which I need to force close JF and restart.

This also randomly happened without caching, just with reverse proxy and apparently I noticed this random issue as far as JF 10.5.x but more noticeable in 10.6.x. Initially I thought it's a bad Wifi connection not till I tested them in the emulator. No issue if I were to point it directly to the binded IP.

I've excluded the following leading paths from caching including the root web socket. Is there any other paths or URIs I should exclude from caching?

```
set $nocache 0;
	if ($request_uri ~* ^/(videos|audio|sessions)/.+) {
		set $nocache 1;
	}
...
proxy_cache_bypass $cookie_nocache $arg_nocache $nocache;
```

Also how can I debug from the emulator to see the requests, warnings or errors? Thank you in advance.

**Comment by dmitrylyzo** [comment]
> Is there any other paths or URIs I should exclude from caching?

At least these are mostly used during playback. Time will tell.

Nginx will response with cached data on identical GET requests, right?
If so, how will this work when Jellyfin refreshes metadata/new items?

> Also how can I debug from the emulator to see the requests, warnings or errors?

That was a long ago. _Trying to reproduce from memory._
1. Uncheck some files and folders (https://github.com/jellyfin/jellyfin-tizen/issues/19#issuecomment-633616422).
2. Run emulator.
3. Select it as a target (it is usually done automatically).
4. Run debug.
It should (re)install app and run web inspector (I think it should also be configured in Tizen Studio).

If you get error `closed closed`, try to convert project type to `tv-samsung (5.5)` (https://github.com/jellyfin/jellyfin-tizen/issues/19#issuecomment-633616422) and rebuild WGT.

**Comment by Nazar78** [comment]
Apologies for the late reply, I was composing this while trying to debug it at the same time. Thanks for the tips, I got the debug running albeit painfully slow despite acceptable workstation specs. Apparently the hanging issue is due to the node_modules.

> At least these are mostly used during playback. Time will tell.

Ok noted thank you.

> Nginx will response with cached data on identical GET requests, right?
> If so, how will this work when Jellyfin refreshes metadata/new items?

Yes it will cache identical GET 200 requests. Although JF is running on SSD and media on RAID 10, I did this to reduce spikes in my NAS resources and it does seem to help a lot also improvements on the speed/load when multiple internal/external clients browsing. I've set the cache duration to 7 days and allocated the default tmpfs 7.8 GB from my total 16 GB RAM but so far it's using only shy 1 GB surprisingly low for over 10 TB of media. I've been monitoring the hit/miss/bypass and seems it's working alright even after metadata/new items gets updated either by schedule, inotify or manually triggered. Probably the hash is different every update but I didn't check further as it's been working good so far.

A little bit more findings on JF Tizen, I suspect after restarting the JF Tizen app, it would somehow forgot the reverse proxy URL and chose to request via the binded IP instead. I thought I saw such issue before being reported in GitHub but I just can't find it. This is because after restarting JF Tizen app, I can no longer see requests in the Nginx logs being made from JF Tizen to the Nginx reverse proxy. To my surprise today (before getting the debug to run) I noticed the request logs in JF are showing local binded IP instead of the reverse proxy domain. This is not the case when accessing from browsers and Android clients. These are the steps taken for both emulator and TV:

1. Launch JF on the TV (newly built from scratch, previous JF app uninstalled prior).

2. Enter the Nginx reverse proxy HTTPS URL then credentials. BTW when using reverse proxy for both emulator and TV I was greeted with this dark text select/add server layout only for the very first launch probably due to the issue. ![image](https://user-images.githubusercontent.com/2883688/89621133-669cdd00-d8c3-11ea-9ca1-711b6fe4aa48.png)

3. Browse around, saw logs being updated in both Nginx reverse proxy and JF.

4. Try to play any video, it will get stuck at loading circle. The request attempt to play video doesn't exist on neither JF or Nginx logs.

5. Force close JF then restart, at this point there's no more requests being made to Nginx reverse proxy as seen in its logs. Checked JF logs I saw local IP instead of the reverse proxy domain in the requests.
> [2020-08-07 15:26:29.814 +08:00] [WRN] [47] Emby.Server.Implementations.HttpServer.HttpListenerHost: HTTP Response 200 to "192.168.1.2". Time (slow): 0:00:00.7687485. "http://192.168.1.6:8096/Items/6408f8817ed2ae04782a3a80aaf55e9d/Images/Primary?maxHeight=320&maxWidth=320&tag=eb33267c8e8dde1818bf2c5949008a66&quality=90"

6. At this point I redo everything again from step 1 but without attempting to play any video. Immediately after entering the domain, credentials and seeing the main screen, I exited the app then restart. After restarted, I only saw two request made in Nginx "GET /Branding/Css" and "GET /system/info/public" nothing else and saw further request such as below log is being made to the binded IP port 8096 instead of the configured HTTPS domain. Resignin again still same issue unless I delete and add again the domain but it will lost it after restart. This has never been an issue with browsers and Android clients.

I've not been monitoring JF logs but it seems my TV has all this while been requesting to the binded IP instead of the configured domain. Are you able to help see if you can replicate these issues especially the video not playing. Think to simplify the setup, the reverse proxy front-end doesn't have to be a domain to test, just different port or IP will suffice.

PS: I got the debug emulator running today and concluded the following for JF Tizen:

a. At the point where the video got stuck loading, as seen from the network console (I cleared the console then attempt to play), there's no request further made by JF Tizen to anywhere not even to the binded IP except for the keepalive. No relevant errors/warnings except for "isUdPanelSupported() error code = undefined" and "focusManager.js:1 Error in focusManager.autoFocus: TypeError: element.focus is not a function".
![image](https://user-images.githubusercontent.com/2883688/89624842-70294380-d8c9-11ea-9b2b-578c8d28fd6b.png)

**Last Nginx entries**:

> xxx.xxx.xxx.xxx - MISS [06/Aug/2020:14:40:24 +0800] "GET /Users/208766dc30734d229d79805c89ac6566/Items/bc23671ee38a1b12fda926c96cfd872f/Intros HTTP/2.0" 200 66 "-" "Mozilla/5.0 (SMART-TV; LINUX; Tizen 5.0) AppleWebKit/537.36 (KHTML, like Gecko) Version/5.0 TV Safari/537.36"
> xxx.xxx.xxx.xxx - - [06/Aug/2020:14:40:24 +0800] "POST /Items/bc23671ee38a1b12fda926c96cfd872f/PlaybackInfo?UserId=208766dc30734d229d79805c89ac6566&StartTimeTicks=0&IsPlayback=true&AutoOpenLiveStream=true&AudioStreamIndex=1&SubtitleStreamIndex=2&MediaSourceId=bc23671ee38a1b12fda926c96cfd872f&MaxStreamingBitrate=193103448 HTTP/2.0" 200 1052 "-" "Mozilla/5.0 (SMART-TV; LINUX; Tizen 5.0) AppleWebKit/537.36 (KHTML, like Gecko) Version/5.0 TV Safari/537.36"
> xxx.xxx.xxx.xxx - MISS [06/Aug/2020:14:40:24 +0800] "GET /Items/bc23671ee38a1b12fda926c96cfd872f/Images/Backdrop?quality=100&tag=fcbba98a168c00060d02ccad36eaff9b HTTP/2.0" 200 119128 "-" "Mozilla/5.0 (SMART-TV; LINUX; Tizen 5.0) AppleWebKit/537.36 (KHTML, like Gecko) Version/5.0 TV Safari/537.36"

**Last JF entires**:
> [2020-08-06 14:40:24.450 +08:00] [INF] [15] MediaBrowser.Api.Playback.MediaInfoService: User policy for "xxx". EnablePlaybackRemuxing: True EnableVideoPlaybackTranscoding: True EnableAudioPlaybackTranscoding: True
> [2020-08-06 14:40:24.479 +08:00] [INF] [15] MediaBrowser.Api.Playback.MediaInfoService: Profile: "Unknown Profile", Path: "/share/Video/test.mp4", isEligibleForDirectPlay: True, isEligibleForDirectStream: True
> [2020-08-06 14:40:24.507 +08:00] [INF] [15] MediaBrowser.Api.Playback.MediaInfoService: RemoteClientBitrateLimit: 15000000, RemoteIp: "xxx.xxx.xxx.xxx", IsInLocalNetwork: False
> [2020-08-06 14:40:24.507 +08:00] [INF] [15] MediaBrowser.Api.Playback.MediaInfoService: Profile: "Unknown Profile", Path: "/share/Video/test.mp4", isEligibleForDirectPlay: True, isEligibleForDirectStream: True
> [2020-08-06 14:40:24.507 +08:00] [INF] [15] MediaBrowser.Api.Playback.MediaInfoService: RemoteClientBitrateLimit: 15000000, RemoteIp: "xxx.xxx.xxx.xxx", IsInLocalNetwork: False
> [2020-08-06 14:40:24.507 +08:00] [INF] [15] MediaBrowser.Api.Playback.MediaInfoService: Profile: "Unknown Profile", Path: "/share/Video/test.mp4", isEligibleForDirectPlay: True, isEligibleForDirectStream: True

b. JF Tizen will permanently revert the request from Nginx reverse proxy domain to the local binded IP after app restart. Can't test this on emulator but the logged requests are sufficient to support its behavior. This issue is seen even without HTTP 200 GET caching on both TV and emulator.

**Comment by dmitrylyzo** [comment]
Saved servers are in `jellyfin_credentials` of localStorage.
I don't have domain but have no-cache proxy. I see two addresses: `ManualAddress` and `LocalAddress`. `LocalAddress` contains direct address (with port). Connection process is here https://github.com/jellyfin/jellyfin-apiclient-javascript/blob/024c0eb7b7634b938f9d485ce25dd57bcb758ab5/src/apiClient.js#L3494-L3505
And it seems to take all available addresses.

I have a "blackened text" too - I think some PR broke it for unauthorized access.

As for playback, that `file` scheme error is likely preventing player from changing the stream.

**Comment by Nazar78** [comment]
> As for playback, that `file` scheme error is likely preventing player from changing the stream.

I can't figure out this `file` scheme :(

I tested with a non-domain reverse proxy (just by using IP and different Nginx front-end listening port, NGINX:8097->JF:8096) and it works perfectly even with or without proxy caching but any Tizen media playback will fail using my https sub-domain.

> I don't have domain but have no-cache proxy

If it's not too much to request, appreciate if you could help test emulator 5.5 with those free DDNS such as https://freedns.afraid.org/ ? Below is my sample config (you can comment out the proxy_cache* and/or ssl* parts) which works with Android and web clients except for Tizen media playback (currently using JF 10.6.3):

```
proxy_cache_path /var/cache/nginx/proxy levels=1:2 use_temp_path=off keys_zone=proxy:10m inactive=7d max_size=7g;

upstream jellyfin {
	server 192.168.1.6:8096;
	keepalive 32;
}

server {
	listen 443 ssl http2;
	ssl_certificate /ssl/fullchain-cert.cer;
	ssl_certificate_key /ssl/cert.key;
	server_name jellyfin.mydomain.com;
	send_timeout 100m;

	set $nocache 0;
	if ($request_uri ~* ^/(videos|audio|sessions)/.+) {
		set $nocache 1;
	}

	proxy_set_header Host $host;
	proxy_set_header X-Real-IP $remote_addr;
	proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
	proxy_set_header X-Forwarded-Proto $scheme;
	proxy_set_header X-Forwarded-Protocol $scheme;
	proxy_set_header X-Forwarded-Host $http_host;
	proxy_buffering off;

	location / {
		proxy_pass http://jellyfin;
		proxy_cache proxy;
		proxy_cache_valid 200 7d;
		proxy_cache_use_stale error timeout invalid_header updating http_500 http_502 http_503 http_504;
		proxy_cache_bypass $cookie_nocache $arg_nocache $nocache;
		proxy_no_cache $cookie_nocache $arg_nocache $nocache;
		proxy_cache_revalidate on;
		proxy_cache_min_uses 1;
		proxy_cache_background_update on;
		proxy_cache_lock on;
		add_header X-Cache-Status $upstream_cache_status;
	}
	location /socket {
		proxy_pass http://jellyfin/socket;
		proxy_http_version 1.1;
		proxy_set_header Sec-WebSocket-Extensions $http_sec_websocket_extensions;
		proxy_set_header Sec-WebSocket-Key $http_sec_websocket_key;
		proxy_set_header Sec-WebSocket-Version $http_sec_websocket_version;
		proxy_set_header Upgrade $http_upgrade;
		proxy_set_header Connection "Upgrade";
	}
}
```

**Comment by dmitrylyzo** [comment]
I've looked where `file` issue occurs:
https://github.com/jellyfin/jellyfin-apiclient-javascript/blob/024c0eb7b7634b938f9d485ce25dd57bcb758ab5/src/appStorage.js#L8
and it was caught, so it probably doesn't break client.
We probably should remove this caching for bundled clients or completely because I don't see reading of `localData` (except checking for change). But I may be wrong.

To verify if `file` issue breaks client or not, you could try this:
1. Clone https://github.com/jellyfin/jellyfin-apiclient-javascript
2. Remove all code related to `localData` and `cache` (in fact, almost all :smile:) in `appStorage.js`
3. `yarn dev` in apiclient source - you will get `dist` folder with modified apiclient.
4. In `jellifyn-web` source, replace `node_modules/jellyfin-apiclient/dist` with freshy built `dist`.
5. `yarn build:development` (to make it more debug frendly) in `jellyfin-web` source.
6. Build WGT as usual starting from `Prepare Interface`.

In addition, you could try to force using of `ManualAddress` (that one from URL line): remove
https://github.com/jellyfin/jellyfin-apiclient-javascript/blob/024c0eb7b7634b938f9d485ce25dd57bcb758ab5/src/connectionManager.js#L620-L631
and
https://github.com/jellyfin/jellyfin-apiclient-javascript/blob/024c0eb7b7634b938f9d485ce25dd57bcb758ab5/src/apiClient.js#L3494-L3497
_Why are there two connects?_ :facepalm:

I still haven't checked domain vs local.

**Comment by Nazar78** [comment]
@dmitrylyzo thank you again for your replies.

> I've looked where `file` issue occurs:
> https://github.com/jellyfin/jellyfin-apiclient-javascript/blob/024c0eb7b7634b938f9d485ce25dd57bcb758ab5/src/appStorage.js#L8
> and it was caught, so it probably doesn't break client.
> We probably should remove this caching for bundled clients or completely because I don't see reading of `localData` (except checking for change). But I may be wrong.
> 
> To verify if `file` issue breaks client or not, you could try this:
> 
> 1. Clone https://github.com/jellyfin/jellyfin-apiclient-javascript
> 2. Remove all code related to `localData` and `cache` (in fact, almost all 😄) in `appStorage.js`
> 3. `yarn dev` in apiclient source - you will get `dist` folder with modified apiclient.
> 4. In `jellifyn-web` source, replace `node_modules/jellyfin-apiclient/dist` with freshy built `dist`.
> 5. `yarn build:development` (to make it more debug frendly) in `jellyfin-web` source.
> 6. Build WGT as usual starting from `Prepare Interface`.
> 
> In addition, you could try to force using of `ManualAddress` (that one from URL line): remove
> https://github.com/jellyfin/jellyfin-apiclient-javascript/blob/024c0eb7b7634b938f9d485ce25dd57bcb758ab5/src/connectionManager.js#L620-L631
> and
> https://github.com/jellyfin/jellyfin-apiclient-javascript/blob/024c0eb7b7634b938f9d485ce25dd57bcb758ab5/src/apiClient.js#L3494-L3497
> _Why are there two connects?_ 🤦

I've tried (blindly without debug; due to my Tizen Studio issue) removing both `localData`/`cache` codes and manually forced `ManualAddress` as you suggested. Every emulator restarts I was asked to re-input the server details so your steps seems correct but the playback attempt still failed with loading circle using reverse proxy domain.

> I still haven't checked domain vs local.

Please let me know anything I can help with this, I can quickly provide you a subdomain pointing to your end.

> you said that Tizen hangs on large files. Have you unchecked unneeded files (`node_modules`) in project properties? [#19 (comment)](https://github.com/jellyfin/jellyfin-tizen/issues/19#issuecomment-633616422)

Yes I've tried everything you mentioned in this link. Previously I've unchecked the `node_modules` as the project was created from repo but now I just import the fresh WGT archive and it's project type to `tv-samsung-5.5`. The hanging will occur almost instantly when the debug progress reached the `www\bundle.js` and if I'm lucky it made past this file, the next huge file will be `www\libraries\subtitles-octopus-worker-legacy.js` running endlessly several hours. When I start interacting with the UI, it will stop responding need to force close.
![image](https://user-images.githubusercontent.com/2883688/91655780-ed6d6000-eae5-11ea-9c11-d0b6ac4eb106.png)

Few times I let it run overnight but end up with exit code 1 error.
![image](https://user-images.githubusercontent.com/2883688/91656270-1d1e6700-eaea-11ea-873b-af9063175bf8.png)

Searched online looks like this is a known issue for Tizen Studio which recommends CLI but how do we debug in CLI? I've tried switching the JDK to the latest 1.8.0_261 as opposed to the default installed by Tizen Studio. I've also tried increasing the Tizen Studio memory to few gigs (16 GB of RAM to spare) `"C:\Devel\Samsung\tizen-studio\ide\eclipse.ini"` what else can I look into?
```
-startup
plugins/org.eclipse.equinox.launcher_1.4.0.v20161219-1356.jar
--launcher.library
plugins/org.eclipse.equinox.launcher.win32.win32.x86_64_1.1.550.v20170928-1359
-nl
en
-vm
javaw.exe
-vmargs
-Dfile.encoding=UTF-8
-Dosgi.requiredJavaVersion=1.6
-XX:+AggressiveOpts
-XX:PermSize=1G
-XX:MaxPermSize=2G
-XX:MaxNewSize=1G
-XX:NewSize=1G
-Xms2g
-Xmx3g
-Dlog4j.configuratorClass=org.tizen.common.util.log.TizenLog4jConfigurator
--add-modules=ALL-SYSTEM
```

**Comment by Nazar78** [comment]
I tried again today debugging the just released JF 10.6.4 imported WGT with a clean installed Tizen Studio 3.7 for Win 64. Still hangs at `www\bundle.js`. Looked at the Tizen Studio IDE logs `"C:\Devel\Samsung\tizen-studio-data\ide\logs\ide-20200831_162347.log"` I only saw this line:

`[2020.08.31 16:30:58][ERROR] PrivilegeCheckUtil.java(245) - Faild to parse "/Jellyfin/www/bundle.js"`

Which Tizen Studio version/OS are you using that debug without issues?

**Comment by dmitrylyzo** [comment]
> Every emulator restarts I was asked to re-input

:confused: It was supposed to use saved `ManualAddress` (automatically).

> Which Tizen Studio version/OS are you using that debug without issues?

3.6 with jre-1.8.0_212 under Kubuntu 19.04
_I know I need to update OS, IDE and build instruction. I am constantly distracted by something more interesting._

> `PrivilegeCheckUtil.java`

Maybe disable some rules?
![Screenshot_20200831_122859](https://user-images.githubusercontent.com/56478732/91706225-c3b84580-eb86-11ea-960e-46a99a29fa72.png)
or
![Screenshot_20200831_124359](https://user-images.githubusercontent.com/56478732/91706872-ab94f600-eb87-11ea-8f0b-6d02d4c8b60f.png)

**Comment by HimbeersaftLP** [comment]
I was able to "reverse-engineer" the debugging procedure using Process Monitor and Wireshark, it's pretty simple to do it using the CLI:
Open a PowerShell Window (or bash on Linux)
Step 1: Transfer the wgt like you normally would
Step 2: `C:\tizen-studio\tools\sdb.exe shell 0 debug "AprZAARz4r.Jellyfin" "300"` (I have no clue what the 300 is for)
Step 3: `& "C:\Program Files (x86)\Google\Chrome\Application\chrome.exe" --no-first-run --activate-on-launch --no-default-browser-check --allow-file-access-from-files --disable-web-security --disable-translate --proxy-auto-detect --enable-blink-features=ShadowDOMV0 --enable-blink-features=CustomElementsV0 --user-data-dir=C:\tizen-studio-data\chrome-user-data --app=http://192.0.0.1:7011/inspector.html?page=1` (instead of 192.0.0.1 put your TV's IP)

**Comment by dmitrylyzo** [comment]
@HimbeersaftLP Thanks for the link.
I am able to connect to emulator - `launch app (debug)` section from link.
What's nice about this method is that app doesn't get reinstalled every time.

**Comment by HimbeersaftLP** [comment]
If your enter key doesn't work in the JS console, the last reply in this thread solves the problem: [link](https://developer.tizen.org/ko/forums/web-application-development/enter-key-suddelny-doesnt-work-web-inspector-console-window-execute-commands.)
1. Create a new folder
2. Run:
```
npm install vitalets/node-webkit-agent-frontend
.\node_modules\.bin\webkit-devtools-agent-frontend
```
3. Open `http://localhost:9090/inspector.html?host=192.0.0.1:7011&page=1` in your favourite web browser.(instead of 192.0.0.1 put your TV's IP)

**Comment by Nazar78** [comment]
@HimbeersaftLP 
Thanks a lot! For my RU8000 Tizen 5.0 it's slightly different though I had to change few things from your suggestion to get it work.

> Step 2: C:\tizen-studio\tools\sdb.exe shell 0 debug "AprZAARz4r.Jellyfin" "300" (I have no clue what the 300 is for)

I can't use the "300", it will return "closed". Without it, I got a reply with the port (seems to be randomly provided). The previous debug session also needs to be ended (exit the app) else it'll just hang:
`... successfully launched pid = 26139 with debug 1 port: <GIVEN-PORT>`

Then as for step 3 Chrome, the port 7011 obviously doesn't work and the URI `/inspector.html?page=1` returned 404 so I ran the root URI with the `<GIVEN-PORT>` instead, then click on the popup `Inspectable WebContents` Jellyfin provided link:
`--app=http://<TV-IP>:<GIVEN-PORT>`


@dmitrylyzo 
> Maybe disable some rules?

I already disabled most if them but thanks for the letting me know your setup, will look into it soon as now the debugging works.

Here's what I did today. The file error doesn't seems to affect anything as you said it's being caught. Still, I commented out the cache part that causes the error in `appStorage.js`:
```
    //if (cache) {
    //    cache.put('data', new Response(JSON.stringify(instance.localData))).catch(onCachePutFail);
    //}
```

I also commented out the `LocalAddress` in `connectionManager.js`:
```
//            if (
//                !serverInfo.manualAddressOnly &&
//                serverInfo.LocalAddress &&
//                addressesStrings.indexOf(serverInfo.LocalAddress) === -1
//            ) {
//                addresses.push({
//                    url: serverInfo.LocalAddress,
//                    mode: ConnectionMode.Local,
//                    timeout: 0
//                });
//                addressesStrings.push(addresses[addresses.length - 1].url);
//            }
```
and `apiClient.js` as you suggested:
```
//    if (serverInfo.LocalAddress && addressesStrings.indexOf(serverInfo.LocalAddress) === -1) {
//        addresses.push({ url: serverInfo.LocalAddress, timeout: 0 });
//        addressesStrings.push(addresses[addresses.length - 1].url);
//    }
```

On top of `yarn build:development`, I also switched to debug logging in JF server and set the Chrome to verbose. When browsing I see my subdomain in the debug/logs but once attempt to play a video I still see it's showing the subdomain but it will fail with endless loading circle.

Below attachments has its IP/Domain/ID filtered:-
Here's the Chrome: 
[3d746079-732d-44c6-9b76-fb727cff71f1.log](https://github.com/jellyfin/jellyfin-tizen/files/5155276/3d746079-732d-44c6-9b76-fb727cff71f1.log)

And the JF log: 
[log_20200901.log](https://github.com/jellyfin/jellyfin-tizen/files/5155277/log_20200901.log)

One thing caught my attention, Chrome requested **HTTPS** but JF logs shows **HTTP**? Still looking into this as I'll need to switch from HTTPS to HTTP domain (301 redirected).

**Comment by BaronGreenback** [comment]
Looks like there is something wrong with you network connecton. Seeing a lot of "[DBG] [38] Emby.Server.Implementations.Networking.NetworkManager: NetworkAvailabilityChanged"

**Comment by Nazar78** [comment]
> Looks like there is something wrong with you network connecton. Seeing a lot of "[DBG] [38] Emby.Server.Implementations.Networking.NetworkManager: NetworkAvailabilityChanged"

Erm I'm not sure about this. Would you kindly help give a hint what kind of issue I could be facing that might be related to the HTTPS domain reverse proxy playback failure? I temporarily added HTTP and it now works for Tizen.

I've looked at my NAS system logs dmesg etc. and it has no apparent network issues. My setup is quite simple. JF server is running in chroot Debian 10 Buster on my NAS bound to its 192.168.1.6 interface.

```
root@Nimbustor4:~ # ip route
default via 192.168.1.1 dev bond0-br
10.8.0.0/24 via 10.8.0.2 dev tun0
10.8.0.2 dev tun0  src 10.8.0.1
10.172.5.0/24 dev nat-br  src 10.172.5.1
172.17.0.0/16 dev docker0  src 172.17.0.1
192.168.1.0/24 dev bond0-br  src 192.168.1.6
```

Then Nginx reverse proxy runs on the same NAS chroot which this NAS has 2 ports doing active load balancing to my router. So it's WAN:443->Nginx:1443->JF:8096. For HTTP, Nginx runs on port 1080. Internet speed test from the NAS is about 900 Mbps both up and down stream. All local (including Tizen TV) and external clients are connecting through the Nginx HTTPS subdomain reverse proxy. There's several off-site Android TV clients connecting to this HTTPS reverse proxy with no issues.

**Comment by BaronGreenback** [comment]
NetworkAvailabilityChange is triggered when the system sees something happening. 
Whereas it shouldn't affect your https reverse proxy playback, you never know with the complexities of JF.

It comes from this object. https://docs.microsoft.com/en-us/dotnet/api/system.net.networkinformation.networkchange.networkavailabilitychanged?view=netcore-3.1

I'm working on the network subsystem - so am looking into all things network.

**Comment by Nazar78** [comment]
It seems the issue is not Tizen specific. After looking at the known chrome cast posted issues and its possible temporary fix, I've managed to run below successfully on Tizen for few weeks.

As opposed to the JF Nginx's documentation and because I'm using proxy cache, I've moved the rest of the "non-socket" `proxy_set_header` outside before the `location` directives. So there's 3 locations, this `API` location code below, `/` and `/socket`.

Hope it can help someone else, I'll close this issue for now.

```
location ~* ^/System/Info/Public$ {
	subs_filter_types application/json;
	subs_filter '{"LocalAddress":"(.+?)"' '{"LocalAddress":"https://jellyfin.mydomain.com/"' r;
	proxy_set_header Accept-Encoding "";
	proxy_pass http://jellyfin;
}
```
